src/Model/Model.ts

```diff
@@ -489,16 +489,33 @@ export class Model<T extends IModelData> implements IModel<T> {
     const service = getDataService(this.serviceName);
     observable
       .takeUntil(this.getWillDestroyObservable$())
-      .subscribe((value) => {
+      .subscribe((updatedValue => {
         if (!this.relatedModels.hasOwnProperty(fieldName)) {
-          this.relatedModels[fieldName] = value;
-        } else if (this.relatedModels[fieldName] !== value) {
-          service
-            .actions
-            .setRelationship({ id: this.id, fieldName, value })
-            .invoke();
+          this.relatedModels[fieldName] = updatedValue;
+        } else if (this.relatedModels[fieldName] !== updatedValue) {
+          const currentValue: Model<any> | Model<any>[] | any = this.relatedModels[fieldName];
+
+          if (currentValue instanceof Array) {
+            if (currentValue.some(related => !related.isDestroying)) {
+              currentValue.forEach(related =>
+                // destroy model, so that the subscription above is canceled
+                related.markForDestruction(),
```

Comment:
I just realized, calling `markForDestruction` on every single item in the array does not make sense. Those individual models themselves have not changed.

I think we should just consider calling `markForDestruction` in each reducer for now.

Reply:
Since the entire array is being replaced (`currentValue` is the old array, `updatedValue` is the new array), all the items in the old array must be marked for destruction.

Reply:
The `currentValue` is an array that is being replaced by `updatedValue` in this case, so all the models in `currentValue` need to be marked for destruction

Reply:
Why? The new array instance could contain items that have not changed, which therefore do not need to get removed.

Watch a few minutes of this YouTube video about immutable architecture:
- https://www.youtube.com/watch?v=rtcn9I9sB5M&feature=youtu.be&t=22m5s
- Specifically, pay close attention to the comments about "structural sharing" and "recycling immutable objects".

By forcing every object to be cleared when they don't need to be will cause performance issues at scale.